src/HotChocolate/Core/src/Types.Scalars/RegularExpressionType.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Text.RegularExpressions;
+using HotChocolate.Language;
+
+namespace HotChocolate.Types.Scalars
+{
+    /// <summary>
+    /// The Regular Expression scalar type represents a String Type following 
+    /// a regular expression pattern/format, as defined
+    /// here https://en.wikipedia.org/wiki/Regular_expression.
+    /// </summary>
+    public class RegularExpressionType : StringType
+    {
+        private readonly string _validationPattern;
+        private readonly Regex _validationRegex; 
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RegularExpressionType"/> class.
+        /// </summary>
+        public RegularExpressionType()
+            : this(
+                WellKnownScalarTypes.RegularExpression,
+                ScalarResources.RegularExpressionType_Description)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RegularExpressionType"/> class.
+        /// </summary>
+        public RegularExpressionType(
+            NameString name,
+            string? description = null,
+            BindingBehavior bind = BindingBehavior.Explicit,
+            string regEx = "" )
```

Comment:
Remove space .... but also agree with @PascalSenn this is obligatory.

```suggestion
            string regEx = "")
```